src/clique.erl

```diff
@@ -95,23 +104,31 @@ run([_Script, "show" | Args] = Cmd) ->
 run([_Script, "describe" | Args] = Cmd) ->
     print(clique_config:describe(Args), Cmd);
 run(Cmd0) ->
-    case is_help(Cmd0) of
-        {ok, Cmd} ->
+    {GlobalFlags, Cmd} = extract_global_flags(Cmd0),
+    case proplists:get_bool(help, GlobalFlags) of
+        true ->
             clique_usage:print(Cmd);
-        _ ->
-            M0 = clique_command:match(Cmd0),
+        false ->
+            Format = proplists:get_value(format, GlobalFlags, "human"),
+            M0 = clique_command:match(Cmd),
             M1 = clique_parser:parse(M0),
             M2 = clique_parser:validate(M1),
-            print(clique_command:run(M2), Cmd0)
+            print(clique_command:run(M2), Cmd0, Format)
     end.
 
-%% @doc Help flags always comes at the end of the command
--spec is_help(iolist()) -> {ok, iolist()} | false.
-is_help(Str) ->
-    [H | T] = lists:reverse(Str),
-    case H =:= "--help" orelse H =:= "-h" of
-        true ->
-            {ok, lists:reverse(T)};
-        false ->
-            false
-    end.
+%% @doc Returns a list of globally applicable flags (e.g. --help), along with
+%% the original command string minus the global flags that were pulled out.
+-spec extract_global_flags([string()]) -> {[{atom(), term()}], [string()]}.
+extract_global_flags(Cmd) ->
+    FoldFun = fun(E, {GlobalAcc, CmdAcc}) ->
+                      case E of
+                          "--format=" ++ Format ->
```

Comment:
So, one issue with doing the parsing this way is that `--format` cannot have a space and then its argument, instead of the equal sign like `riak-admin blah --format csv`. This is possible with all other (non-global) flags and results in an inconsistent behavior. It's also the reason `clique_parser:parse_flags/3` is so complex. I think we can do one of two things here: Either not allow spaces between flags and arguments and remove that support or fix this to allow that behavior. I personally like the flexibility and it matches what getopt supports which is why I implemented it that way.

Note that key/value arguments for a given command (non-flags) cannot have spaces between them because the parser uses the = sign to determine what is an argument to the command so it can do an exact match lookup of the command name in ets. It's evil to allow spaces between k/v arguments that aren't flags if we ever want to allow arguments that aren't k/v in the same command as k/v args. That's possible with a change to the parsing code so that it relies on command prefix longest match search instead of a k-v lookup in ets.